lib/services/appmanager-service.ts

```diff
@@ -50,14 +52,52 @@ class AppManagerService implements IAppManagerService {
 			let projectName = this.$project.projectData.ProjectName;
 			let solutionPath = buildResult[0].solutionPath;
 			let projectPath = solutionPath.substr(solutionPath.indexOf("/") + 1);
-			this.$server.tam.uploadApplication(projectName, projectName, projectPath).wait();
 
-			this.$logger.info("Successfully uploaded package.");
+			let publishSettings: Server.PublishSettings = {
+				IsPublished: options.publish,
+				NotifyByPush: options.sendPush,
+				NotifyByEmail: options.sendEmail,
+				Groups: []
+			};
+
+			if (options.group.length) {
+				publishSettings.Groups = this.findGroups(options.group).wait();
+			}
 
+			this.$server.tam.uploadApplication(projectName, projectName, projectPath, publishSettings).wait();
+			this.$logger.info("Successfully uploaded package.");
 			this.openAppManagerStore();
 		}).future<void>()();
 	}
 
+	public getGroups(): IFuture<void> {
+		return ((): any  => {
+			this.$project.ensureProject();
```

Comment:
https://github.com/Icenium/icenium-cli/blob/master/lib/project.ts#L52 
injecting the $project will not cause any errors, as Project's constructor tries to read project data, but if you are not inside project, it does not fail, it just continues (for cases like current). So wherever you need project, you should call $project.ensureProject() or $project.ensureCordovaProject() (when you have to be sure the project is hybrid)